packages/azure-http-specs/specs/azure/resource-manager/resources/mockapi.ts

```diff
@@ -151,17 +163,137 @@ const validResourceExtensionsResource = {
   },
 };
 
+function getServiceKey(resourceUri: string) {
+  switch (resourceUri) {
+    case RESOURCE_GROUP_SCOPE_URI:
+      return "ResourceGroup" as const;
+    case SUBSCRIPTION_SCOPE_URI:
+      return "Subscription" as const;
+    case TENANT_SCOPE_URI:
+      return "Tenant" as const;
+    case RESOURCE_SCOPE_URI:
+      return "Resource" as const;
+    default:
+      throw new ValidationError(
+        "Invalid resource uri",
+        `${RESOURCE_GROUP_SCOPE_URI} | ${SUBSCRIPTION_SCOPE_URI} | ${TENANT_SCOPE_URI} | ${RESOURCE_SCOPE_URI}`,
+        resourceUri,
+      );
+  }
+}
+
+function requestHandler(
+  req: MockRequest,
+  requestMethod: string,
+  resourceUri: string,
+  validResource: any,
+) {
+  const serviceKey = getServiceKey(resourceUri);
+  switch (requestMethod) {
+    case "get":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json(validResource),
+      } as const;
+    case "put":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json(validResource),
+      } as const;
+    case "patch":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json({
+          ...validResource,
+          properties: {
+            provisioningState: "Succeeded",
+            description: "valid2",
+          },
+        }),
+      } as const;
+    case "delete":
+      return {
+        pass: serviceKey,
+        status: 204,
+      } as const;
+    case "list":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json({
+          value: [validResource],
+        }),
+      } as const;
+    default:
+      throw new ValidationError(
+        "Invalid request method",
+        `"get" | "put" | "patch" | "delete" | "list"`,
+        requestMethod,
+      );
+  }
+}
+
+function getUri(resourceUri: string, extensionName: string, requestMethod: string): string {
+  switch (requestMethod) {
+    case "list":
+      return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`;
+    default:
+      return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/${extensionName}`;
+  }
+}
+
+function getUriForTenant(
```

Comment:
Why a separate method for tenant? 
Will `getUri("/", extensionName, requestMethod)` + `getUri("", extensionName, requestMethod)` work for tenant resource?

Reply:
Fixed by the [commit#22089c1](https://github.com/Azure/typespec-azure/pull/2203/commits/22089c1d3e99d0d62f04e5f3f7560e034608c143)